FFMpegCore.Extensions.Downloader/FFMpegDownloader.cs

```diff
@@ -0,0 +1,276 @@
+﻿using System.IO.Compression;
+using System.Net;
+using System.Runtime.InteropServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace FFMpegCore.Extensions.Downloader;
+
+/// <summary>
+///     Downloads the latest FFMpeg suite binaries from ffbinaries.com.
+/// </summary>
+public class FFMpegDownloader
+{
+    [Flags]
+    public enum FFMpegBinaries : ushort
+    {
+        FFMpeg,
+        FFProbe,
+        FFPlay
+    }
+
+    public enum FFMpegVersions : ushort
```

Comment:
You could use the C# [Description] attribute here so instead of building a hard coded dictionary of FFMpegVersions:strings you could just pull the description from the enum and that is the path to download that version